File: HelloFunctionGradle/src/main/kotlin/com/czwl/kotlin/types/classes/java/Person.java
package com.czwl.kotlin.types.classes.java;

/**
 * @author CZ
 * @version 1.0
 * @date 2020/2/1 11:23
 */
public class Person {

    private int age;//field
    private String name;

    public Person(int age, String name) {
        this.age = age;
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        System.out.println("setAge"+ age);
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: HelloFunctionGradle/src/main/kotlin/com/czwl/kotlin/types/classes/kotlin/Person.kt
package com.czwl.kotlin.types.classes.kotlin

/**
 * @author  CZ
 * @date  2020/2/1 11:29
 * @version 1.0
 */
class Person (age:Int,name:String){
    var age:Int=age//property
    get(){
        return field
    }
    set(value) {
        println("setAge:$value")
        field=value

    }
    var name:String=name
    get() {return field}//backing field
    set(value){

    }
}
fun main(){
    val  ageRef=Person::age
    val person=Person(18,"W")
    val nameRef=person::name
    nameRef.set("L")
    ageRef.set(person,19)
}
